### PR TITLE
Reduce DDC SyntheticEvent warning noise

### PR DESCRIPTION
**Pull request by greglittlefield-wf**
## Motivation
As a workaround to https://github.com/dart-lang/sdk/issues/43939, our `SyntheticEvent` classes are non `@anonymous`.

A consequence of this is that, in the DDC, when event handlers are type-checked in certain situations, they emit console warnings every time.

These warnings can quickly become excessive when interacting with an app.
Screen Shot 2020-12-16 at 4 44 15 PM

## Solution
- In dev builds of React, wrap console.warn and only emit errors for SyntheticEvent wrappers once
    - Also log an additional message indicating that the warning is expected, and linking to the relevant Dart issue
- Add tests

Boyscouting
- Update outdated name of js_interop_helpers_test test directory and outdated README

## Testing
- Verify tests pass in both DDC and dart2js
    - DDC (run `pbr test -- -P dartdevc`)
    - dart2js (run `pbr test -r -- -P dart2js`)
- Verify that the tests properly test this behavior
    - Revert compiled JS changes (commit e2fe5a4)
   - Verify that `console.warn` tests fail in DDC (`pbr test -- -P dartdevc`) for only the "dev" suites (prod files should not be affected)
- Pull this branch into an app where this issue occurs and verify that the error is only logged once
    -  e.g., run the w_context_menu web directory and hover/select items in the context menu

### CODE REVIEW

File: test/js_builds/shared_tests.dart [hunk 1]
```diff
@@ -0,0 +1,147 @@
+// ignore_for_file: deprecated_member_use_from_same_package
```
Comment (greglittlefield-wf) [inline test/js_builds/shared_tests.dart:1]: This file looks like it's all new and the other file was deleted, but it was just renamed with content added.

I strongly recommend reviewing [all commits up to the rename commit](https://github.com/cleandart/react-dart/pull/293/files/e2fe5a4465bbd738c6a3e4e2e5241f4a90f2280a) separately from [the rename commit itself](https://github.com/cleandart/react-dart/pull/293/commits/1692efb1d089654bc9159435ae5aa387a4ad6a27) for a better diff.

File: js_src/react.js [hunk 2]
```diff
@@ -6,6 +6,9 @@ import 'core-js/es/set';
 import 'core-js/stable/reflect/delete-property';
 import 'core-js/stable/object/assign';
 
+// Used by dart_dev_dev.js
```
Comment (aaronlademann-wf) [inline js_src/react.js:9]: #supernit 

```suggestion
// Used by dart_env_dev.js
```

**Review by aaronlademann-wf — APPROVED** [review]
+1

**Review by aaronlademann-wf — APPROVED** [review]
+1

**Comment by sydneyjodon-wk** [comment]
QA +1

- [x] Verify tests pass in both DDC and dart2js
    - [x] DDC (run `pbr test -- -P dartdevc`)
    - [x] dart2js (run `pbr test -r -- -P dart2js`)
- [x] Verify that the tests properly test this behavior
   - [x] Revert compiled JS changes (commit e2fe5a4)
   - [x] Verify that `console.warn` tests fail in DDC (`pbr test -- -P dartdevc`) for only the "dev" suites (prod files should not be affected)
- [x] Pull this branch into `w_context_menu` and `w_input_validation` and verify that the error is only logged once

**Pull request merged**